### PR TITLE
added tests

### PR DESCRIPTION
**Pull request by KarimKhalil33**
## 📝 Description

Added comprehensive test coverage for code analysis utilities to ensure reliability and maintainability of the codebase. This PR includes:

- **Updated existing tests** in `test_code_analysis.py` to cover newly added functions (`analyze_github_commit_patterns`, `infer_roles_from_commit_files`, `_detect_frameworks`)
- **New test suite** for text processing utilities (`test_text_processing.py`) covering camelCase splitting, filename keyword extraction, and keyword ranking
- **New test suite** for user preferences utilities (`test_user_preferences.py`) covering preference loading, keyword weighting, and resume enhancement
- **Enhanced edge case handling** to properly test None inputs and malformed data scenarios

All tests are focused and concise (under 300 lines total) with proper mocking for external dependencies and comprehensive coverage of both happy path and error scenarios.

**Closes:** #481 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

**Test Coverage Added:**

- [x] **Text Processing Tests** - 4 test functions covering all utilities (`split_camelcase_and_filter`, `extract_meaningful_filename_keywords`, `get_top_keywords`)
- [x] **User Preferences Tests** - 5 test functions with proper mocking for external dependencies
- [x] **Updated Code Analysis Tests** - Added coverage for newly implemented functions
- [x] **Edge Case Testing** - Comprehensive handling of None inputs, empty data, and malformed structures

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [x] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

</details>

### CODE REVIEW

**Review by kjassani — APPROVED** [review]
Thanks for adding this comprehensive test suite Karim! All pass on my end
Image

Image

File: tests/test_code_analysis.py [hunk 1]
```diff
@@ -126,11 +127,50 @@ def test_new_json_structure_compatibility():
...
+    test_files = [
+        {"path_after": "frontend/app.js"}, 
+        {"path_after": "backend/models.py"}, 
+        {"path_after": "tests/test.py"}
+    ]
+    roles = infer_roles_from_commit_files(test_files)
+    print_subsection("Inferred Roles", roles)
+    assert isinstance(roles, set)
+
 def test_edge_cases_and_robustness():
```
[the remark this answers is not in the document]
Comment (kjassani) [reply tests/test_code_analysis.py]: I was also getting a failing test in that file, but when I ran the test inside the Docker container, it passed successfully.
